packages/rest/src/parser.ts

```diff
@@ -101,13 +100,31 @@ function buildOperationArguments(
   body?: MaybeBody,
 ): OperationArgs {
   const args: OperationArgs = [];
-
+  let requestBodyIndex: number = -1;
+  if (hasArgumentsFromBody(operationSpec)) {
+    const i = (<RequestBodyObject>operationSpec.requestBody)[
+      REQUEST_BODY_INDEX
+    ];
+    requestBodyIndex = i ? i : 0;
+  }
+  if (
```

Comment:
I am refactoring the `requestBody` insert function to make the logic neater then these.